File: src/app/interface/invoice-payment.ts
export interface InvoicePayment {
  id?: number;
  invoiceId: number;
  paymentDate: string;
  amount: number;
  paymentMethod: string;
  referenceNumber?: string;
  notes?: string;
  createdBy?: number;
  createdByName?: string;
  createdAt?: string;
  updatedAt?: string;
  invoiceNumber?: string;
  clientName?: string;
}

export interface PaymentAnalytics {
  startDate: string;
  endDate: string;
  totalPayments: number;
}

File: backend/src/main/java/com/bostoneo/bostoneosolutions/model/InvoicePayment.java
package com.bostoneo.bostoneosolutions.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class InvoicePayment {
    private Long id;
    private Long invoiceId;
    private LocalDate paymentDate;
    private BigDecimal amount;
    private String paymentMethod;
    private String referenceNumber;
    private String notes;
    private Long createdBy;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    // Relations
    @JsonIgnore
    private Invoice invoice;
    private User createdByUser;
}